### PR TITLE
refactor: build resource-group image in repo

### PR DESCRIPTION
**Pull request by sdowell**
This changes the resource-group-controller image to be built alongside the other first party images rather than built separately and imported as a third party image. The ResourceGroup controller is a core component of Config Sync, and this is intended to simplify the maintenance of this component.

### CODE REVIEW

File: cmd/resource-group/main.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 Google LLC
```
Comment (nan-yu) [inline cmd/resource-group/main.go:1]: nit: 2023

File: cmd/resource-group/main.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"os"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
```
Comment (nan-yu) [inline cmd/resource-group/main.go:20]: I think I missed that in the resource-group repo, but the import is no longer needed.

File: cmd/resource-group/main.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	"kpt.dev/resourcegroup/controllers/runner"
+	// +kubebuilder:scaffold:imports
```
Comment (nan-yu) [inline cmd/resource-group/main.go:22]: same here

File: cmd/resource-group/main.go (hunk 2 above)
Comment (sdowell) [reply cmd/resource-group/main.go]: done

File: cmd/resource-group/main.go (hunk 3 above)
Comment (sdowell) [reply cmd/resource-group/main.go]: done

File: cmd/resource-group/main.go (hunk 1 above)
Comment (sdowell) [reply cmd/resource-group/main.go:1]: done

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**